dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,345 @@
+* All TxOut scripts shall be version 0.
+* TxOut[0] script must be OP_TADD
+* TxOut[1] is optional if there is no change. If there is change it shall be a default OP_SSTXCHANGE script.
+
+{|
+!Transaction!!Script!!Description
+|-
+|TxIn[0..N]||Default utxo scripts||Normal utxo scripts as you would find on normal transactions
+|-
+|TxOut[0]||[0]OP_TADD||Value shall be what is credited to the treasury
+|-
+|TxOut[1]||[0]OP_SSTXCHANGE||Optional if there is no change.
+|}
+
+Additional consensus enforced characteristics:
+* A TAdd transaction shall not have a zero value for change.
+* There shall be no more than twenty TAdd transactions per block.
+
+===Debit treasury===
+
+The TSpend transactions are rather complex and some additional definitions need
+to be made prior to explaining the format.
+
+Required chain configuration parameters:
+* Treasury Vote Interval, or TVI, determines in which blocks a TSpend may occur. A block is considered on a TVI when the block height modulo TVI is equal to zero.
+* Treasury Vote Interval Multiplier, or TVIM, dictates the total duration (total duration = TVI * TVIM) where votes for or against the expenditure are considered valid.
+* Treasury Vote Quorum Multiplier and Divisor, TVQM and TVQD respectively, are used to calculate the required number of votes to reach quorum.
+* Treasury Vote Required Multiplier and Divisor, TVRM and TVRD respectively, are used to calculate the required number of Yes votes.
+* Treasury Expenditure Policy, or TEP, determines the overall duration where all treasury expenditures are summed and averaged.
+* Treasury Expenditure Window, or TEW, is used to calculate the window that is used to verify that the sum of all TSpends does not exceed ~150% of the average treasury spending.
+* Treasury Expenditure Bootstrap, or TEB, is used to determine a baseline expenditure in order to bootstrap treasury payouts.
+* Politeia Keys, or PiKeys, are a number of well-known Schnorr public keys that are allowed to sign TSpend transactions.
+
+Only the Pi key administrator can author transactions that debit the treasury
+account and generate utxos for recipients.
+
+The life-cycle of a TSpend transaction is as follows
+# PiKey administrator creates a TSpend transaction with an Expiry in the future.
+# The TSpend is signed and broadcast to the network.
+# If the TSpend is nominally valid (Expiry in the future, valid signature, valid public key etc) it is allowed in the mempool.
+# Once in the TSpend hits the mempool wallets will be notified.
+# Voting wallets with a voting policy will now start voting either yes/no or abstain at the TSpend vote window start.
+# Votes are accumulated every block and once they reach an insurmountable tally they are short circuited.
+# At the end of the TSpend window (which is always on a TVI because start is zero based) votes are tallied and if quorum and yes votes breach their respective thresholds the TSpend transaction is added to the block.
```

Comment:
```suggestion
# At the end of the TSpend window (which is always on a TVI because start is zero based) votes are tallied. If quorum and yes votes breach their respective thresholds the TSpend transaction is added to the block.
```

Reply:
I feel like the and really belongs there. If anyone else trips over it I'll redo the sentence.